File: from-0401-to-0500/0451-sort-characters-by-frequency/solution.py
class Solution(object):
    def frequencySort(self, s):
        """
        :type s: str
        :rtype: str
        """
        if not s:
            return s
        
        import collections
        freqs = collections.defaultdict(int)
        for c in s:
            freqs[c]+=1
        char_items = sorted(list(freqs.items()), key=lambda item:item[1], reverse=True)
        result = []
        for item in char_items:
            result.extend([item[0]] * item[1])
            
        return ''.join(result)

File: from-0401-to-0500/0451-sort-characters-by-frequency/solution.go
func frequencySort(s string) string {
    freq := make(map[byte]int)
    for i := 0; i < len(s); i ++ {
        freq[s[i]] += 1
    }
    items := [][]int{}
    for c, f := range freq {
        items = append(items, []int{int(c), f})
    }
    sort.Slice(items, func(i, j int) bool {
        return items[i][1] > items[j][1]
    })
    result := []byte{}
    for _, item := range items {
        for i := 0; i < item[1]; i++ {
            result = append(result, byte(item[0]))
        }
    }
    return string(result)
}